File: nav-frontend/src/types/searchEngine.ts
import type { EntityId } from './common'

export interface SearchEngine {
  id: EntityId
  name: string
  icon?: string | null
  searchUrl: string
  placeholder?: string | null
  isDefault: boolean
  sortOrder: number
}

export interface AdminSearchEngine extends SearchEngine {
  visible: boolean
  createdAt?: string
  updatedAt?: string
}

export interface SearchEnginePayload {
  name: string
  icon: string
  searchUrl: string
  placeholder: string
  sortOrder: number
  visible: boolean
}

File: nav-backend/src/main/java/com/example/nav/module/search/entity/SearchEngine.java
package com.example.nav.module.search.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@TableName("search_engine")
public class SearchEngine {

    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private String icon;
    private String searchUrl;
    private String placeholder;
    @TableField("is_default")
    private Boolean defaultEngine;
    private Integer sortOrder;
    private Boolean visible;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
